polkadot/xcm/xcm-builder/src/fee_handling.rs

```diff
@@ -100,7 +101,7 @@ pub struct XcmFeeToAccount<AssetTransactor, AccountId, ReceiverAccount>(
 
 impl<
 		AssetTransactor: TransactAsset,
-		AccountId: Clone + Into<[u8; 32]>,
+		AccountId: Clone + IntoLocation,
```

Comment:
```suggestion
		DestLocation: Get<Location>,
```

Looks better to me instead of requiring a new trait.

Reply:
Makes sense, will ipdate the code soon!

Reply:
Tried to update the code accordingly but encountered the following problem:
`get()` does not take `self` as a parameter in any form. And we need the address of the given account in order to convert it to `xcm::v4::Junction::AccountId32 { network: None, id: ??? }`. So, I think with the API for `Get`, it is impossible for this case to convert an `AccountId` to `Location` with `Get`.

However, I looked at the code of `XcmFeeToAccount` and `deposit_or_burn_fee`, and `AccountId` is not used anywhere for the business logic. It is just there as an intermediate step to convert the provided `ReceiverAccount` into `Location`. 

So, we can just change this:
```rust
		AccountId: Clone + IntoLocation,
		ReceiverAccount: Get<AccountId>,
```

to this:
```rust
		ReceiverAccount: Get<Location>,
```

and there won't be any intermediate `AccountId`. 

This will be a breaking change though... Hence, `IntoLocation` still seems like the way to go to me eyes in terms of backwards compatibility.

Please let me know if I missed anything 👍

Reply:
> and there won't be any intermediate AccountId.

Yeah that sounds more reasonable.

> This will be a breaking change though... Hence, IntoLocation still seems like the way to go to me eyes in terms of backwards compatibility.

`IntoLocation` is also a breaking change. We will maybe be happy that no code breaks because everybody is using `AccountId32`, but this not really given.

We could also just introduce a new type and let this type here with the old interface and set it as deprecated.

Reply:
Yes, the cleanest way will be deprecating that. Will update the code accordingly for the new type

Reply:
Done! I didn't spend much time on the namings, and I'm open to suggestions.